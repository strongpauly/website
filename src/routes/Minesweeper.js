import React, { Component } from 'react';
import HiFrame from '../components/HiFrame';

export class Minesweeper extends Component {

  render() {
    return <div id="minesweeperContainer">
      <div className="container">
        An implementation of the classic game using react. It&apos;s source can be found <a href="https://www.github.com/strongpauly/minesweeper">here</a>.
      </div>
      <HiFrame src="/minesweeper" title="minesweeper"></HiFrame>
    </div>;
  }

}
