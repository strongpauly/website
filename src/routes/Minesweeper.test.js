import React from 'react';
import {Minesweeper} from './Minesweeper';
import {shallow} from 'enzyme';

/* eslint-env jest */

describe('Minesweeper', () => {
  it('renders without crashing', () => {
    const app = shallow(<Minesweeper/>);
    expect(app).toMatchSnapshot();
  });

  it('creates a #minesweeperContainer', () => {
    const app = shallow(<Minesweeper/>);
    expect(app.find('#minesweeperContainer')).toHaveLength(1);
  });
});
