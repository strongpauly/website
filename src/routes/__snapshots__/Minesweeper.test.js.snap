// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Minesweeper renders without crashing 1`] = `
<div
  id="minesweeperContainer"
>
  <div
    className="container"
  >
    An implementation of the classic game using react. It's source can be found 
    <a
      href="https://www.github.com/strongpauly/minesweeper"
    >
      here
    </a>
    .
  </div>
  <HiFrame
    src="/minesweeper"
    title="minesweeper"
  />
</div>
`;
